dvc/command/pkg.py

```diff
@@ -54,36 +119,89 @@ def add_parser(subparsers, parent_parser):
         formatter_class=argparse.RawDescriptionHelpFormatter,
     )
     pkg_install_parser.add_argument(
-        "address",
-        nargs="?",
-        default="",
-        help="Package address: git://<url> or https://github.com/...",
+        "targets", nargs="*", default=[None], help="Package name."
     )
-    pkg_install_parser.add_argument(
-        "target_dir",
-        metavar="target",
-        nargs="?",
-        default=".",
-        help="Target directory to deploy package outputs. "
-        "Default value is the current dir.",
+    pkg_install_parser.set_defaults(func=CmdPkgInstall)
+
+    PKG_UNINSTALL_HELP = "Uninstall package(s)."
+    pkg_uninstall_parser = pkg_subparsers.add_parser(
+        "uninstall",
+        parents=[parent_config_parser, parent_parser],
+        description=append_doc_link(PKG_UNINSTALL_HELP, "pkg-uninstall"),
+        help=PKG_UNINSTALL_HELP,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
     )
-    pkg_install_parser.add_argument(
-        "-s",
-        "--select",
-        metavar="OUT",
-        action="append",
-        default=[],
-        help="Select and persist only specified outputs from a package. "
-        "The parameter can be used multiple times. "
-        "All outputs will be selected by default.",
+    pkg_uninstall_parser.add_argument(
+        "targets", nargs="*", default=[None], help="Package name."
     )
-    pkg_install_parser.add_argument(
-        "-f",
-        "--file",
-        help="Specify name of the stage file. It should be "
-        "either 'Dvcfile' or have a '.dvc' suffix (e.g. "
-        "'prepare.dvc', 'clean.dvc', etc). "
-        "By default the file has 'mod_' prefix and imported package name "
-        "followed by .dvc",
+    pkg_uninstall_parser.set_defaults(func=CmdPkgUninstall)
+
+    PKG_ADD_HELP = "Add package."
+    pkg_add_parser = pkg_subparsers.add_parser(
+        "add",
+        parents=[parent_config_parser, parent_parser],
+        description=append_doc_link(PKG_ADD_HELP, "pkg-add"),
+        help=PKG_ADD_HELP,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
     )
-    pkg_install_parser.set_defaults(func=CmdPkgInstall)
+    pkg_add_parser.add_argument("name", help="Package name.")
```

Comment:
I noticed you can send all sorts of weird strings to `name` including possibly problematic strings and characters like `\n`. Maybe it would be best to limit to alphanumeric + dashes (something like regex `^[a-zA-Z][a-zA-Z0-9\-_]?`)?

Reply:
I also noticed if you provide an existing name, the package is replaced silently. This could lead into confusion if it has already been installed since the contents in the installed package won't necessarily match those of the URL in the `pkg` config file section, which may have been replaced alter package installation.

Reply:
> I noticed you can send all sorts of weird strings to name including possibly problematic strings and characters like \n or those starting with numbers. Maybe it would be best to limit to alphanumeric + dashes (something like regex ^[a-zA-Z][a-zA-Z0-9\-_]?)?

Good point. The problem is more generic that `pkg`, so I've created https://github.com/iterative/dvc/issues/2091 for now.

> I also noticed if you provide an existing name, the package is replaced silently. This could lead to confusion if it has already been installed since the contents in the installed package won't necessarily match those of the URL in the pkg config file section, which may have been replaced alter package installation.

Fixed. Thanks! :slightly_smiling_face:

Reply:
Thank you.